src/pages/EditRequestPage.js

```diff
@@ -104,6 +125,35 @@ function EditRequestPage({report, route, policy, policyCategories, policyTags, p
         });
     }, [parentReportAction, fieldToEdit]);
 
+    const updateTaxAmount = useCallback(
+        (transactionChanges) => {
+            const newTaxAmount = CurrencyUtils.convertToBackendAmount(Number.parseFloat(transactionChanges.amount));
+
+            if (newTaxAmount === TransactionUtils.getTaxAmount(transaction)) {
+                Navigation.dismissModal();
+                return;
+            }
+            IOU.updateMoneyRequestTaxAmount(transaction.transactionID, report.reportID, newTaxAmount, policy, policyTags, policyCategories);
+            Navigation.dismissModal(report.reportID);
+        },
+        [transaction, report, policy, policyTags, policyCategories],
+    );
+
+    const updateTaxRate = useCallback(
+        (transactionChanges) => {
+            const newTaxCode = transactionChanges.data.code;
+
+            if (!newTaxCode) {
```

Comment:
This doesn't seem to be working. Like taxAmount, don't we need to compare with the transaction code?

Reply:
T'was working this way, but I've made condition more explicit. We can test this again.